File: DESIGN-CSharp/SimpleFactory/Adventurer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SimpleFactory
{
    class Adventurer : Archer
    {
        public override string getType()
        {
            Console.WriteLine("this is Adventurer ");
            return this.GetType().Name;
        }
    }
}

File: D1/SimpleFactory/Adventurer.py
class Adventurer:
    def getType(self):
        pass

class Warrior(Adventurer):
    def getType(self):
        print("This is Warrior")
        return type(self).__name__

class Archer(Adventurer):
    def getType(self):
        print("This is Adventurer")
        return type(self).__name__